File: solutions/6kyu/BitCounting.java
/*
* kata link
* https://www.codewars.com/kata/526571aae218b8ee490006f4
* */

public class BitCounting {
    
    public static int countBits(int n){
        int res = 0;
        while( n > 0 ){
            if(n % 2 != 0)
                res++;
            n /= 2;
        }
        return res;      
    }
    
}

File: solutions/6kyu/bit_counting.c
/*
* kata link
* https://www.codewars.com/kata/526571aae218b8ee490006f4
* */

#include <stddef.h>

size_t bit_counting(unsigned v)
{
  int b = 0;
  while(v > 0){
    b += v%2;
    v = v / 2;
  }
  return b;
}